### PR TITLE
#165434232 fix onchange descriptive errors

### PR DESCRIPTION
**Pull request by joelethan**
### What does this PR do?
- This PR shows descriptive error messages on the signup page as the user inputs data 

### Description of Task to be completed?
- Provide the user with a signup form
- Provide descriptive error messages that disappear when validations pass

### How should this be manually tested?
- Clone the [repo](https://github.com/andela/ah-aquaman-frontend) and cd into ah-aquaman-frontend
- checkout the branch `bg-fix-onchange-errors-165434232`
- Run the commands below to install packages and run the server
          - npm install
          - npm start
- Navigate to `/signup` 
- Provide `username`, `email` and `password` to register
- View how error messages disappear when validations pass

### What are the relevant pivotal tracker stories?

- [#165434232](https://www.pivotaltracker.com/story/show/165434232)

### Screenshots
Screenshot 2019-04-18 at 11 47 34

### CODE REVIEW

**Review by Peace-Apple — APPROVED** [review]
Great work @joelethan, the deployment works fine and the feedback has been implemented.

**Review by CryceTruly — APPROVED** [review]

**Review by BagzieGracious — APPROVED** [review]
Thank you @joelethan for your good work, we have reviewed the work and you implemented the feedback

**Review by wasibani-roy — APPROVED** [review]
Good work @joelethan

File: src/views/signupView/index.js [hunk 1]
```diff
@@ -31,6 +38,58 @@ export class SignupView extends Component {
...
+          }, 
+        });
+        return true;
+      }
+      this.setState({
+        validateInput: {
+          username: false,
+        }, 
+      });
+      return false;
```
Comment (AmosWels) [inline src/views/signupView/index.js:68]: Does this return setState ?

File: src/views/signupView/index.js [hunk 2]
```diff
@@ -31,6 +38,58 @@ export class SignupView extends Component {
...
+    }
+    if (e.target.name === "password1") {
+      if (e.target.value.length > 0 && e.target.value.length < 9) {
+        this.setState({ password1: e.target.value });
+        this.setState({
+          validateInput: {
+            password: true,
+          }, 
+        });
+        return true;
```
Comment (AmosWels) [inline src/views/signupView/index.js:78]: same here

**Review by AmosWels — CHANGES_REQUESTED** [review]
Goodwork @joelethan , Check that you resolve the lint issues with the npm fix.

**Review by AmosWels — APPROVED** [review]

**Pull request merged**